rs/nns/governance/src/neuron_store/metrics.rs

```diff
@@ -176,18 +186,18 @@ pub(crate) struct NeuronSubsetMetrics {
 impl NeuronSubsetMetrics {
     fn increment(
         &mut self,
-        voting_power_economics: &VotingPowerEconomics,
         now_seconds: u64,
         neuron: &Neuron,
+        // This is somewhat expensive. Therefore, instead of recalculating it a
+        // few times, we require that the caller calculate it.
+        pre_calculated_deciding_voting_power: u64,
     ) {
         let staked_e8s = neuron.minted_stake_e8s();
         let staked_maturity_e8s_equivalent =
             neuron.staked_maturity_e8s_equivalent.unwrap_or_default();
         let maturity_e8s_equivalent = neuron.maturity_e8s_equivalent;
 
         let potential_voting_power = neuron.potential_voting_power(now_seconds);
-        let deciding_voting_power =
```

Comment:
This is where deciding voting power was re-calculated. To avoid this, caller must pass pre-calculated deciding voting power.